### PR TITLE
Task 25

### PR DESCRIPTION
**Pull request by miyabitanimchi**
@pmvald 
It is my first task, so please let me know if there are any big/small mistakes or my misunderstandings . 
Thank you.

### CODE REVIEW

**Comment by miyabitanimchi** [comment]
Oh, I'm sorry, I think I misunderstood about the branch name. I should've created one with the task number on my local branch and pushed that branch? I made a branch with my name instead...

**Comment by miyabitanimchi** [comment]
Hi @j-ae ! 
Oh I got it, thank you so much! 🙏

File: -sass/_styles.m.scss [hunk 1]
```diff
@@ -20,6 +20,8 @@
 @import "styles/utilities";
 @import "styles/footers.m";
 @import "styles/iboxes";
+@import "styles/sections.m";
+@import "styles/action-sheet.m";
```
Comment (pmvald) [inline -sass/_styles.m.scss:24]: Good

File: -sass/styles/_action-sheet.m.scss [hunk 2]
```diff
@@ -0,0 +1,29 @@
+.s-action-sheet {
+  h2 {
+    text-align: center;
+    font-size: 0.8125rem;
```
Comment (pmvald) [inline -sass/styles/_action-sheet.m.scss:4]: how about $font-size-sm? (bootstrap has several scss variables related to font size, pelase check them in bootstrap5/_variables. scss, I guess we should use font-size-sm or something similar to make the text one level smaller)

File: -sass/styles/_action-sheet.m.scss [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    color: $gray-900;
```
Comment (pmvald) [inline -sass/styles/_action-sheet.m.scss:5]: For colors lets use the them colors defined in _styles/_varibales, like maybe $grey or $light-grey?

File: -sass/styles/_action-sheet.m.scss [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    padding: 0.375*$spacer;
+    border-bottom: 1px solid $gray-300;
```
Comment (pmvald) [inline -sass/styles/_action-sheet.m.scss:7]: same here, $grey-300 -> $light-grey/$grey/$dark-grey ...

File: -sass/styles/_action-sheet.m.scss [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  section {
+    border-bottom: 1px solid $gray-300;
+    .menu-column {
+      display: flex;
+      flex-direction: column;
+      align-items: start;
```
Comment (pmvald) [inline -sass/styles/_action-sheet.m.scss:15]: no need. The `<menu>` tag and its modifier (`menu--column` i guess) are already defined in `/styles/_menu` and contain all these styles.

File: -sass/styles/_action-sheet.m.scss [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+      li {
+        padding: 0.625*$spacer $spacer;
+      }
+      a {
+        font-size: 0.9375rem;
```
Comment (pmvald) [inline -sass/styles/_action-sheet.m.scss:21]: $font-size-sm

File: -sass/styles/_action-sheet.m.scss [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      }
+    }
+
+    .close-item {
+      color: $red;
```
Comment (pmvald) [inline -sass/styles/_action-sheet.m.scss:26]: $red? no problem just make sure it is defined in `/styles/_variables` (or add it if not)

File: -sass/styles/_sections.m.scss [hunk 8]
```diff
@@ -0,0 +1,33 @@
+.s-section {
+  background: $light-grey;
```
Comment (pmvald) [inline -sass/styles/_sections.m.scss:2]: good

File: -sass/styles/_sections.m.scss [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+  h2 {
+    padding: 1.125*$spacer $spacer 0.375*$spacer $spacer;
+    font-size: 0.9375rem;
+    color: $gray-900;
```
Comment (pmvald) [inline -sass/styles/_sections.m.scss:7]: see above comments

File: -sass/styles/_sections.m.scss [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  .s-section__content {
+    background: $white;
```
Comment (pmvald) [inline -sass/styles/_sections.m.scss:11]: just white

File: -sass/styles/_sections.m.scss [hunk 11]
```diff
@@ -0,0 +1,33 @@
+.s-section {
+  background: $light-grey;
+
+  h2 {
+    padding: 1.125*$spacer $spacer 0.375*$spacer $spacer;
+    font-size: 0.9375rem;
```
Comment (pmvald) [inline -sass/styles/_sections.m.scss:6]: h2 normally should be *bigger* than the normal size font ...

File: -sass/styles/_sections.m.scss [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+      margin: $spacer 0;
+      font-size: 0.8125rem;
+      border-radius: 6px;
+      cursor: pointer;
+
+      &:hover {
+      background: #566376;
+      }
+    }
+  }
```
Comment (pmvald) [inline -sass/styles/_sections.m.scss:27]: what is content__btn? We only have __content section and no need to add any style to anything inside it. This has to be removed

File: -sass/styles/_sections.m.scss [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+  
+  p {
+    color: $gray-800;
+  }
```
Comment (pmvald) [inline -sass/styles/_sections.m.scss:31]: remove

File: tasks/task-025/action-sheet.html [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+        </menu>
+      </section>
+
+      <section>
+        <menu class="menu-column">
+          <li><a class="close-item"> Close </a></li>
+        </menu>
+      </section>
+
+    </article>
```
Comment (pmvald) [inline tasks/task-025/action-sheet.html:27]: Good

**Review by pmvald — CHANGES_REQUESTED** [review]
@miyabitanimchi Good, I made some suggestion. Please attend

File: -sass/styles/_action-sheet.m.scss (hunk 2 above)
Comment (miyabitanimchi) [reply -sass/styles/_action-sheet.m.scss:4]: @pmvald Okay for the variable. As you mentioned below, this also shouldn't be h2?

**Review by pmvald — APPROVED** [review]

**Comment by pmvald** [comment]
@miyabitanimchi I made some changes and merged the branch. Now your follow-up assignment: what changes I made? And why? (particularly see the changes related to the menu)

**Comment by miyabitanimchi** [comment]
@pmvald 
Thank you for modifying. I checked the changes and my answers are below.
<br/>

### [action-sheet.html / scss]
1.  **close link's style:**
**Changes:** external → inline, color #dc3545 as a variable → red
**My answers:** That variable with the specific color will never be reused to other element and for the color, just pure red was fine. 

2. **font-size**
**Changes:** font-size-sm → font-size-base
**My answers:** Too small. And also for `<h2>`, it shouldn't be smaller than the normal font size.

3. **border color of section**
**Changes:** $grey-300 → $light-grey
**My answers:** I'm sorry, I forgot to fix it. 

4. **scss - nested li and a in section**
**Changes:** li → menu > li,  a → & > a
**My answers:**Since it's possible that other elements will be added in the section tag,  It's better to write with more specific ways. For  parent > first child, style will be applied only the first child of its parent.

### [sections.html / scss]
1.  **background color of s-section:**
**Changes:** $light-grey → #f9f9f9

2.  **font-size and color of h2:**
**Changes:** font-size-lg → font-size-base, color: $dark-grey → black
**My answers:** For the size, too big? For the color, black is easier for users to read since they are titles for each section.

3.  **Way to write a nested element:**
**Changes:** s-section__content → &__content
**My answers:**  If it's sass, we can write like this and it allows us to read scss files easier and more simple. 

4.  **border-top / bottom on s-section__content:**
**Changes:** border-top and bottom are added
**My answers:**  It makes users easily recognize a border of the title areas and the content areas. 

<br/>
I have questions:

1. For the style of the close link on action-sheet, is there any reason that you used inline css? Are there any rules for when you should use inline or external?

2. For the background of s-section, you changed the color from $light-grey to #f9f9f9. You didn't make a variable for this because that color would be used only on this file?

<br/>
Thank you, and also let me know if my answers are wrong.

**Comment by miyabitanimchi** [comment]
@pmvald 
Hi, I would like to take this task.

**Comment by pmvald** [comment]
@miyabitanimchi Good! For the close link style: we don't need such style, this task is only for a general action sheet. The user can change the text color using other utility classes (like `s-mute`, `s-red`, `s-bold`, etc.) We have to MINIMIZE the number of classes and REUSE the classes as much as possible. For the section color, change, yes, it's a good idea to make it into a varibale on top of the page, something like `$section-bg-color`, but right now I'm thinking maybe we just keep the sections all transparent and just change the body background color ... I'll have to double check some standard themes and see if this is a good idea ...

**Pull request merged**